Domain/Common/ValueObject.cs

```diff
@@ -0,0 +1,47 @@
+﻿namespace Domain.Common
+{
+    // Learn more: https://docs.microsoft.com/en-us/dotnet/standard/microservices-architecture/microservice-ddd-cqrs-patterns/implement-value-objects
+    public abstract class ValueObject
+    {
+        protected static bool EqualOperator(ValueObject left, ValueObject right)
+        {
+            if (left is null ^ right is null)
+            {
+                return false;
+            }
+
+            return left?.Equals(right!) != false;
+        }
+
+        protected static bool NotEqualOperator(ValueObject left, ValueObject right)
+        {
+            return !(EqualOperator(left, right));
+        }
+
+        protected abstract IEnumerable<object> GetEqualityComponents();
+
+        public override bool Equals(object? obj)
+        {
+            if (obj == null || obj.GetType() != GetType())
+            {
+                return false;
+            }
+
+            var other = (ValueObject)obj;
+            return GetEqualityComponents().SequenceEqual(other.GetEqualityComponents());
+        }
+
+        public override int GetHashCode()
+        {
+            var hash = new HashCode();
+
+            foreach (var component in GetEqualityComponents())
+            {
+                hash.Add(component);
+            }
+
+            return hash.ToHashCode();
+        }
```

Comment:
i think it's better to let every (value object) child instance to override the  GetHashCode() to
guarantee that instances of object with the same values will produce the same hash code because HashCode struct may not guarantee consistent hash codes across different executions of the program.
@Majed-404